### PR TITLE
Improve git hooks

### PR DESCRIPTION
**Pull request by karol-bisztyga**
I noticed that for pre-commit hooks, all files are processed.
I think we should process only the staged files since they're the ones that are going to be in the commit.
I think we shouldn't check the files that are not going to be in the commit.

### CODE REVIEW

**Review by kasperski95 — APPROVED** [review]

File: pre-commit [hunk 1]
```diff
@@ -1,5 +1,23 @@
 #!/bin/sh
 
-poetry run poe type_check
-poetry run poe format_check
-poetry run poe lint
+set -e
+
+STAGED_FILES=$(git diff --name-only --cached)
+STAGED_FILES_FOR_FORMAT=$(poetry run poe show_format_check_path | tail -n 1 | xargs git diff --name-only --staged)
+STAGED_FILES_FOR_LINT=$(poetry run poe show_lint_path | tail -n 1 | xargs git diff --name-only --staged)
```
[the remark this answers is not in the document]
Comment (karol-bisztyga) [reply pre-commit:7]: Yes, this is correct, I'm going to rebase to that PR and try to figure it out.

[the remark this answers is not in the document]
Comment (karol-bisztyga) [reply pre-commit:7]: Resolving, newest changes take care of this

File: scripts/run_format_check.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+find protostar -name '*.py' | xargs black --check
```
Comment (karol-bisztyga) [inline scripts/run_format_check.sh:3]: I couldn't find a way of doing this without a separate script, `pyproject.toml` seems to be a bit retarded when it comes to running bash instructions.

File: scripts/run_lint.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+find . -name '*.py' | xargs pylint
```
Comment (karol-bisztyga) [inline scripts/run_lint.sh:3]: I couldn't find a way of doing this without a separate script, `pyproject.toml` seems to be a bit retarded when it comes to running bash instructions.

Also, with the changes added in #1292, the CLI output was very polluted by poetry which printed out all affected files (they make a pretty big list).

File: pre-commit (hunk 1 above)
Comment (karol-bisztyga) [inline pre-commit:7]: oops, this needs fixing :/

Comment (karol-bisztyga) [inline pre-commit:7]: oops, this needs fixing

**Comment by karol-bisztyga** [comment]
I saw some bug, lemme fix that

**Comment by karol-bisztyga** [comment]
I think it can be reviewed now

File: scripts/get_code_check_paths.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+set -e
+
+if [ "$#" != 1 ]; then
+  echo "Invalid number of arguments, expected 1: lint/format";
+  exit 1;
+fi;
+
+if [ "$1" == "lint" ]; then
+  find . -name '*.py'
```
Comment (kasperski95) [inline scripts/get_code_check_paths.sh:10]: I've just run it locally, and the linter complained about files in the virtual environment (.venv). Maybe we can duplicate source of paths in favor of simplicity?

Comment (karol-bisztyga) [reply scripts/get_code_check_paths.sh]: Nice catch. First, I thought we could use paths specified in gitignore to exclude them from the search but it looked too hacky and weird. I think specifying the paths explicitly is simpler and clearer.

Comment (karol-bisztyga) [reply scripts/get_code_check_paths.sh]: [explicitly specify paths to look for python files](https://github.com/software-mansion/protostar/pull/1295/commits/8e6123f43cbf50d807047a164327afc6256b6713)

**Review by kasperski95 — APPROVED** [review]

**Pull request merged**